[LTC] Upstream utils to extract BackendDevice from at::Tensor

alanwaketan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #70069
* #70066

Summary:
This commit upstreams utils to extract BackendDevice from at::Tensor.

Test Plan:
./build/bin/test_lazy --gtest_filter=BackendDeviceTest.GetBackendDevice*

Differential Revision: [D33293160](https://our.internmc.facebook.com/intern/diff/D33293160)

Commit:
[LTC] Upstream utils to extract BackendDevice from at::Tensor

Summary:
This commit upstreams utils to extract BackendDevice from at::Tensor.

Test Plan:
CI

[ghstack-poisoned]

alanwaketan commented:
@alanwaketan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33172665).

wconstab commented on torch/csrc/lazy/backend/backend_device.cpp:
```diff
@@ -53,5 +54,16 @@ c10::Device backendDeviceToAtenDevice(const BackendDevice& device) {
+}
+
+c10::optional<BackendDevice> GetBackendDevice() {
+  return c10::nullopt;
```
What is the point of this function?

wconstab replied:
nevermind i see the comment in the header.

alanwaketan replied:
Great!

Commit:
Update on "[LTC] Upstream utils to extract BackendDevice from at::Tensor"


Summary:
This commit upstreams utils to extract BackendDevice from at::Tensor.

Test Plan:
WIP

Differential Revision: [D33172665](https://our.internmc.facebook.com/intern/diff/D33172665)

[ghstack-poisoned]

alanwaketan commented:
Thanks, Will.

alanwaketan commented:
@alanwaketan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33172665).

Commit:
Update on "[LTC] Upstream utils to extract BackendDevice from at::Tensor"


Summary:
This commit upstreams utils to extract BackendDevice from at::Tensor.

Test Plan:
WIP

Differential Revision: [D33172665](https://our.internmc.facebook.com/intern/diff/D33172665)

[ghstack-poisoned]

alanwaketan commented:
@alanwaketan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33172665).

Commit:
Update on "[LTC] Upstream utils to extract BackendDevice from at::Tensor"


Summary:
This commit upstreams utils to extract BackendDevice from at::Tensor.

Test Plan:
./build/bin/test_lazy --gtest_filter=BackendDeviceTest.GetBackendDevice*

Differential Revision: [D33172665](https://our.internmc.facebook.com/intern/diff/D33172665)

[ghstack-poisoned]

alanwaketan commented:
@alanwaketan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33172665).

suo commented:
Looks like this broke windows builds. http://hud2.pytorch.org/commit/121d06799954b0c0ba24f084f2d78b1a81a4e4ec I am reverting

Commit:
Update on "[LTC] Upstream utils to extract BackendDevice from at::Tensor"


Summary:
This commit upstreams utils to extract BackendDevice from at::Tensor.

Test Plan:
./build/bin/test_lazy --gtest_filter=BackendDeviceTest.GetBackendDevice*

Differential Revision: [D33172665](https://our.internmc.facebook.com/intern/diff/D33172665)

[ghstack-poisoned]

alanwaketan commented:
@alanwaketan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33293160).
